src/main/java/chatbot/Parser.java

```diff
@@ -0,0 +1,205 @@
+package chatbot;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+/**
+ * The Parser class handles user interactions and processes input commands.
+ * It contains logic to interpret user commands and to determine the appropriate response
+ * or error message, based on the current state and data of the application.
+ */
+
+public class Parser {
+
+  public Parser() {}
+
+  /**
+   * Checks if the user wants to end the chat.
+   *
+   * @param words Array of strings representing the user's input, split into words.
+   * @return true if the first word is "bye", indicating a request to end the chat; false otherwise.
+   */
+  public boolean isItDone(String[] words) {
+    return words[0].equals("bye");
+  }
+
+  /**
+   * Processes the user input and performs actions based on the given commands.
+   * This method can add, delete, or modify tasks in the task list, mark tasks as done or not done,
+   * and handle other user commands. It returns a string response that should be shown to the user.
+   * In case of invalid or incomplete commands, it throws specific exceptions.
+   *
+   * @param io The input string provided by the user.
+   * @param words Array of strings representing the user's input, split into words.
+   * @param detail The detail part of the user's input, excluding the command word.
+   * @param myList The current list of tasks.
+   * @param st The storage handler for reading from and writing to the data file.
+   * @return A string response based on the user's input and the action performed.
+   * @throws EmptyTodoException If a "todo" command is given without a description.
+   * @throws EmptyDeadlineException If a "deadline" command is given without a description or deadline.
+   * @throws EmptyEventException If an "event" command is given without a description or event time.
+   * @throws InvalidException If an invalid command is given.
+   */
```

Comment:
Great javadocs usage,  very clearly explained all the parameters and errors thrown.